Brooks/New.hs

```diff
@@ -0,0 +1,484 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module New (
```

Comment:
Not a good name...

Reply:
For the record, this file exists in Drasil already (in the code/ directory). It's equivalent to the current AST.hs, but it doesn't really define an AST since we switched to finally tagless. Would Symantics.hs be a good name?